### PR TITLE
feat(scanners) : ignore files from scanning using mimetype

### PR DESCRIPTION
**Pull request by ag4ums**
Signed-off-by: Anupam Ghosh <anupam.ghosh@siemens.com>

## Description

Sometime scanner ( like nomos takes longtime to finish),
Idea is to skip files with particular mimetypes those not require eg : .class , image, doc, binary  

### Changes

New field in Admin-->Customize --> "Skip MimeTypes from scanning"
   -- to add mimetypes of the files need to be included 

User can decide to include or not to include the mimetype/s included in Admin-->customized page 
(using the checkbox "Ignore SCM files (Git, SVN, TFS) and files with particular Mimetype" in upload page) 

## How to test

1. Add mimetypes that need to be excluded by scanner(now only for nomos) 
in Admin-->Customize --> "Skip MimeTypes from scanning
1. check the checkbox on upload page "Ignore SCM files (Git, SVN, TFS) and files with particular Mimetype"
1. include nomos from scanner list
(https://help.github.com/articles/closing-issues-using-keywords/)

### CODE REVIEW

**Comment by ag4ums** [comment]
(from discussion )
move scanners file selection query to library, to maintain uniformity across all the scanners( nomos, monk, ojo, atarashi, copyrights and ecc ).

**Comment by ag4ums** [comment]
Now ignoring files with particular Mimetypes works with all the agents : 
* nomos
* monk
* ojo
* copyright
* ecc

Also this can be schedule from the  Jobs--> Schedule Agent

File: src/ojo/agent/OjoState.cc [hunk 1]
```diff
@@ -59,16 +59,16 @@ const OjoAgent& OjoState::getOjoAgent() const
  * @param verbosity Verbosity set by CLI
  * @param json      True to get output in JSON format
```
Comment (GMishx) [inline src/ojo/agent/OjoState.cc:60]: Please update the parameter list.

File: src/ojo/agent/OjoState.hpp [hunk 2]
```diff
@@ -36,12 +36,14 @@ class OjoCliOptions
   private:
     int verbosity;  /**< The verbosity level */
     bool json;      /**< Whether to generate JSON output */
+    bool ignoreFilesWithMimeType; /* exclude files as defined in mimetype */
```
Comment (GMishx) [inline src/ojo/agent/OjoState.hpp:39]: Please fix the docs comment syntax.

File: src/ojo/agent/OjoUtils.cc [hunk 3]
```diff
@@ -110,10 +110,10 @@ void bail(int exitval)
  * @return True in case of successful scan, false otherwise.
```
Comment (GMishx) [inline src/ojo/agent/OjoUtils.cc:110]: Please update the parameter list.

File: src/ojo/agent/OjosDatabaseHandler.cc [hunk 4]
```diff
@@ -38,32 +38,9 @@ OjosDatabaseHandler::OjosDatabaseHandler(DbManager dbManager) :
  * @param uploadId Upload ID to be queried
```
Comment (GMishx) [inline src/ojo/agent/OjosDatabaseHandler.cc:38]: Please update the parameter list.

**Pull request merged**